File: app/src/main/java/com/hqq/core/example/ui/data/DataBindingActivity.java
package com.hqq.core.example.ui.data;

import android.widget.TextView;

import androidx.lifecycle.Observer;

import com.hqq.core.example.R;
import com.hqq.core.ui.BaseActivity;
import com.hqq.core.utils.log.LogUtils;

import butterknife.BindView;
import butterknife.OnClick;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.core.app.ui.databing
 * @FileName :   DataBindingActivity
 * @Date : 2019/10/22 0022  下午 1:37
 * @Email :  qiqiang213@gmail.com
 * @Descrive : DataBinding 与现在的设计冲突
 */
public class DataBindingActivity extends BaseActivity {

    @BindView(R.id.textView2)
    TextView mTextView2;

    @Override
    public int setViewId() {
        return R.layout.activity_data_binding;
    }

    @Override
    public void initView() {

        LiveUser.getInstance(this).observe(this, new Observer<User>() {
            @Override
            public void onChanged(User user) {
                LogUtils.e("onChanged        " + user.toString());
            }
        });

    }


    @OnClick(R.id.button19)
    public void onViewClicked() {
        User user = LiveUser.getInstance(this).getValue();
        if (null == user) {
            user = User.newInstance();
        }
        user.setLevel(user.getLevel() + 2);
        LiveUser.getInstance(this).setValue(user);

    }
}

File: app/src/main/java/com/easy/example/ui/jetpack/databinding/DataBindingActivity.kt
package com.easy.example.ui.jetpack.databinding

import android.app.Activity
import android.content.Intent
import android.view.View
import android.widget.LinearLayout
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.easy.core.ui.base.BaseActivity
import com.easy.example.BR
import com.easy.example.R
import com.easy.example.ui.jetpack.livedata.User

/**
 * @Author : huangqiqiang
 * @Package : com.easy.example.ui.dagger
 * @FileName :   DaggerAActivity
 * @Date : 2020/7/15 0015  下午 4:18
 * @Email :  qiqiang213@gmail.com
 * @Descrive : dataBinding  双向绑定 需要实现BaseObservable 进行观察 以及通知回调
 * 注意 setVariable
 */
class DataBindingActivity : BaseActivity() {
    companion object {
        fun open(context: Activity) {
            val starter = Intent(context, DataBindingActivity::class.java)
            context.startActivityForResult(starter, -1)
        }
    }


    override val layoutViewId: Int = R.layout.activity_dagger_a

    override fun initView() {
        val viewDataBinding = DataBindingUtil.inflate<ViewDataBinding>(layoutInflater, R.layout.item_data_binding, null, false)
        (findViewById<View>(R.id.ll_rootView) as LinearLayout).addView(viewDataBinding.root)
        val user = com.easy.example.ui.jetpack.livedata.User.newInstance()
        viewDataBinding.lifecycleOwner = this
        viewDataBinding.setVariable(BR.user, user)
        viewDataBinding.setVariable(BR.dataBinding, DataBanding())
        user.name = user.name + "M"
    }


}